### PR TITLE
Extract panda3d renderer

### PR DESCRIPTION
**Pull request by Gamenot**

### CODE REVIEW

File: smarts/core/renderer_base.py [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+
+    SPAM = 1
+    DEBUG = 2
+    INFO = 3
+    WARNING = 4
+    ERROR = 5
+
+
+class RendererBase:
+    """The base class for renderering
```
Comment (saulfield) [inline smarts/core/renderer_base.py:43]:
```suggestion
    """The base class for rendering
```

**Review by saulfield — APPROVED** [review]

File: smarts/p3d/renderer.py [hunk 2]
```diff
@@ -55,25 +55,16 @@
...
-from .masks import RenderMasks
-from .scenario import Scenario
+from smarts.core import glsl, models
+from smarts.core.colors import Colors, SceneColors
+from smarts.core.coordinates import Pose
+from smarts.core.masks import RenderMasks
+from smarts.core.renderer_base import DEBUG_MODE, RendererBase
+from smarts.core.scenario import Scenario
 
 # pytype: enable=import-error
```
Comment (Adaickalavan) [inline smarts/p3d/renderer.py:65]: Why is this `# pytype: enable=import-error` here? Perhaps it can be removed.

Comment (Gamenot) [reply smarts/p3d/renderer.py:65]: It used to be related to the Panda3d imports.

File: smarts/core/renderer_base.py [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+    def remove_vehicle_node(self, vid: str):
+        """Remove a vehicle node"""
+        raise NotImplementedError
+
+    def camera_for_id(self, camera_id) -> "RendererBase.OffscreenCamera":
+        """Get a camera by its id."""
+        raise NotImplementedError
+
+    @dataclass
+    class OffscreenCamera:
```
[the remark this answers is not in the document]
Comment (Gamenot) [reply smarts/core/renderer_base.py:117]: Steve was a fan of this. It does, however, seem unnecessary.

**Pull request merged**